x-pack/plugins/security/server/authorization/privileges/feature_privilege_iterator/sub_feature_privilege_iterator.ts

```diff
@@ -4,14 +4,19 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { LicenseType } from '../../../../../licensing/server';
+
 import { KibanaFeature, SubFeaturePrivilegeConfig } from '../../../../../features/common';
 
 export function* subFeaturePrivilegeIterator(
-  feature: KibanaFeature
+  feature: KibanaFeature,
+  licenseType: LicenseType
 ): IterableIterator<SubFeaturePrivilegeConfig> {
   for (const subFeature of feature.subFeatures) {
     for (const group of subFeature.privilegeGroups) {
-      yield* group.privileges;
+      yield* group.privileges.filter(
+        (privilege) => !privilege.minimumLicense || privilege.minimumLicense <= licenseType
+      );
```

Comment:
TIL we are still using generators 😅

Reply:
Should we not be using generators anymore?

Reply:
Yeah, I'm curious as well. We use them in quite a few places in security and I love them! :stuck_out_tongue:

Reply:
Generators are fine. Using them for synchronous logic where `map/reduce/filter/flatMap` would be able to generate an array as (more?) easily seems overkill to me, but that's only personal preference, don't mind this comment :)